File: src/main/java/com/kelsos/mbrc/commands/ProtocolRequest.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.model.SocketMessage;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.net.Protocol;
import com.kelsos.mbrc.net.SocketService;

public class ProtocolRequest implements ICommand {
    private SocketService socket;

    @Inject public ProtocolRequest(SocketService socket) {
        this.socket = socket;
    }

    public void execute(IEvent e) {
        socket.sendData(new SocketMessage(Protocol.Protocol, Protocol.Request, 2));
    }
}

File: app/src/main/java/com/kelsos/mbrc/commands/ProtocolRequest.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.networking.client.SocketMessage
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.Protocol
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage
import com.kelsos.mbrc.networking.protocol.ProtocolPayload
import javax.inject.Inject

class ProtocolRequest
  @Inject
  constructor(
    private val socket: SocketService,
  ) : ProtocolAction {
    override fun execute(message: ProtocolMessage) {
      val payload = ProtocolPayload()
      payload.noBroadcast = false
      payload.protocolVersion = Protocol.PROTOCOL_VERSION_NUMBER
      socket.sendData(SocketMessage.create(Protocol.PROTOCOL_TAG, payload))
    }
  }
